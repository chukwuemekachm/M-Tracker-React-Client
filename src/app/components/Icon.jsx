import React from 'react';
import PropTypes from 'prop-types';
import '../assets/css/icon.css';

/**
 * @description The Icon component
 *
 * @param {string} title The title of the icon
 * @param {object} icon The icon image/svg
 *
 * @returns {object}
 */
const Icon = ({ title, icon }) => (
  <div className="col-md-3 text-center">
    <div className="icon-container">
      <img src={icon} className="about-img" alt="authenticate" />
    </div>
    <h4>
      {title}
    </h4>
  </div>
);

Icon.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};

export default Icon;
