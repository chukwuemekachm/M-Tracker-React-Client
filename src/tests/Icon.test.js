import React from 'react';
import { shallow } from 'enzyme';
import Icon from '../app/components/Icon';

describe('Footer component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<Icon title="Repair" icon="icon icon-md" />);
    expect(wrapper.find('div')).toHaveLength(2);
    expect(wrapper.find('img')).toHaveLength(1);
    expect(wrapper.find('h4')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
