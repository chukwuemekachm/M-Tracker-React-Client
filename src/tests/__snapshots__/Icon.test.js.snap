// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render without errors 1`] = `
<div
  className="ch-col-3"
>
  <i
    className="icon icon-md"
  />
  <h3>
    Repair
  </h3>
</div>
`;
